tests/h/views/api_groups_test.py

```diff
@@ -100,12 +100,74 @@ def open_groups(self, factories):
         return [factories.OpenGroup(), factories.OpenGroup()]
 
     @pytest.fixture
-    def anonymous_request(self, pyramid_request):
-        pyramid_request.user = None
+    def authenticated_request(self, pyramid_request, factories):
+        pyramid_request.user = factories.User()
         return pyramid_request
 
+
+@pytest.mark.usefixtures('CreateGroupAPISchema',
+                         'group_service',
+                         'GroupContext',
+                         'GroupJSONPresenter')
+class TestCreateGroup(object):
+
+    def test_it_inits_group_create_schema(self, pyramid_request, CreateGroupAPISchema):  # noqa: N803
+        views.create(pyramid_request)
+
+        CreateGroupAPISchema.assert_called_once()
+
+    # @TODO Move this test once _json_payload() has been moved to a reusable util module
+    def test_it_raises_if_json_parsing_fails(self, pyramid_request):
+        """It raises PayloadError if parsing of the request body fails."""
+        # Make accessing the request.json_body property raise ValueError.
+        type(pyramid_request).json_body = {}
+        with mock.patch.object(type(pyramid_request),
+                               'json_body',
+                               new_callable=mock.PropertyMock) as json_body:
+            json_body.side_effect = ValueError()
+            with pytest.raises(views.PayloadError):
+                views.create(pyramid_request)
+
+    def test_it_proxies_to_group_create_service(self, pyramid_request, group_service):
+        views.create(pyramid_request)
+
+        group_service.create_private_group.assert_called_once()
+
+    def test_it_passes_request_params_to_group_create_service(self,  # noqa: N803
+                                                              pyramid_request,
+                                                              CreateGroupAPISchema,
+                                                              group_service):
+        CreateGroupAPISchema.return_value.validate.return_value = {'name': 'My Group'}
+        views.create(pyramid_request)
+
+        group_service.create_private_group.assert_called_once_with('My Group', pyramid_request.user.userid)
+
+    def test_it_creates_group_context_from_created_group(self,   # noqa: N803
+                                                         pyramid_request,
+                                                         GroupContext,
+                                                         group_service,
+                                                         factories):
+        my_group = factories.Group()
+        group_service.create_private_group.return_value = my_group
+
+        views.create(pyramid_request)
+
+        GroupContext.assert_called_with(my_group, pyramid_request)
```

Comment:
Does `group_service.create_private_group.return_value` need to be a `models.Group` rather than a mock for this test?

```python
    def test_it_creates_group_context_from_created_group(self,
                                                         pyramid_request,
                                                         GroupContext,
                                                         group_service):
        views.create(pyramid_request)

        GroupContext.assert_called_with(
            group_service.create_private_group.return_value, pyramid_request)
```